File: OC演练代码/OC演练代码/ViewController.m
//
//  ViewController.m
//  OC演练代码
//
//  Created by lichuanjun on 2017/5/23.
//  Copyright © 2017年 lichuanjun. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    [self demoSwitch];
    
    UILabel *l = [[UILabel alloc] initWithFrame:CGRectMake(0, 20, 100, 40)];
    l.text = @"hello";
    [self.view addSubview:l];
    
    {
        UILabel *l = [[UILabel alloc] initWithFrame:CGRectMake(0, 60, 100, 40)];
        l.text = @"world";
        [self.view addSubview:l];
    }
    
    // 扩展知识点：要能看懂，在一些老的代码中会有
    // {} 包装代码，() 是表示执行
    // 工作中，看到的最多的表现代码是 UITableView
    UILabel *myLabel = ({
        
        UILabel *l = [[UILabel alloc] initWithFrame:CGRectMake(0, 200, 100, 40)];
        l.text = @"hello world";
        [self.view addSubview:l];
        
        // 最后的 l 会给myLabel 设置数值
        l;
    });
    
    NSLog(@"%@",myLabel);
    
    
    UITableView *tv = ({
        UITableView *tableView = [[UITableView alloc] initWithFrame:CGRectZero style:UITableViewStylePlain];
        
        // ...
        
        tableView;
    });
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


- (void)demoSwitch {
    /**
     1. 分支值类型必须是整数
     2. 每个语句都需要一个 break
     3. 如果要穿透
     4. 如果要定义句柄变量，需要{}
     */
    int num = 10;
    switch (num) {
        case 9:
        case 10:
        {
            NSString *name = @"zhang";
            
            NSLog(@"优 %@",name);
        }
            break;
            
        default:
            NSLog(@"一般");
            break;
    }
}

@end

File: 006-逻辑分支/006-逻辑分支/ViewController.swift
//
//  ViewController.swift
//  006-逻辑分支
//
//  Created by lichuanjun on 2017/5/21.
//  Copyright © 2017年 lichuanjun. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        demo1()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    // MARK: - 三目
    func demo1() -> () {
        let x = 3
        
        // 三目的语法和OC一样的
        x > 5 ? print("大了") : print("小了")
        
        // 分享，有印象就行
        // `()` 表示空执行
        x > 5 ? print("大了") : ()
    }
    
    // MARK: - 简单的分支
    func demo() -> () {
        let x = 10
        
        /**
         1. 条件不需要 ()
         2. 语句必须有 {}
            - 很多公司的代码规范中，禁止不使用 {}
        */
        if x > 5 {
            print("大了")
        } else {
            // Will never be executed
            // 永远不会被执行 - 编译器编译的时候，会做语法检查！
            print("小了")
        }
        
    }
}

